ingestion/src/metadata/ingestion/source/pipeline/databrickspipeline/metadata.py

```diff
@@ -86,25 +91,38 @@ def create(
             )
         return cls(config, metadata)
 
-    def get_pipelines_list(self) -> Iterable[dict]:
-        for workflow in self.client.list_jobs() or []:
-            yield workflow
+    def get_pipelines_list(self) -> Iterable[DataBrickPipelineDetails]:
+        try:
+            for workflow in self.client.list_jobs() or []:
+                yield DataBrickPipelineDetails(**workflow)
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(f"Failed to get pipeline list due to : {exc}")
+        return None
+
+    def get_pipeline_name(self, pipeline_details: DataBrickPipelineDetails) -> str:
```

Comment:
this can return none
```suggestion
    def get_pipeline_name(self, pipeline_details: DataBrickPipelineDetails) -> Optional[str]:
```